### ISSUE TITLE
Provide an alternative for users without Flash

### ISSUE THREAD

**Issue by bmintz** [zeroclipboard/zeroclipboard#552]
I think that users who visit sites that use ZeroClipboard are left behind.
Perhaps ZeroClipboard should detect if Flash is available before anything else. If the user-agent doesn't, ZeroClipboard should do something similar to what https://goo.gl does—select the text that needs copying, then prompting the user to copy it themselves, like so:
![goo.gl's simulated "copy to clipboard"](https://cloud.githubusercontent.com/assets/538336/6425118/dcfd1126-bedf-11e4-8f99-d2edc194e9b1.png).

**Comment by JamesMGreene** [comment]
ZeroClipboard prides itself on leaving the UI up to the consumer, so I definitely do not plan to integrate such a thing.  It would already be fairly easy to do with yourself by adding handlers for the various `"error"` events that ZeroClipboard emits.

That said, I think your point is also strong enough that it warrants perhaps a more useful event so that consumers don't need to listen to and distinguish between all of the various `"error"` events.  For example, perhaps an event named `"fallback"`?

**Comment by JamesMGreene** [comment]
Related to my [comment](https://github.com/zeroclipboard/zeroclipboard/issues/552#issuecomment-77400029): twbs/bootstrap#16269

**Comment by tarebyte** [comment]
I agree with @JamesMGreene, in a lot of use cases it would be really nice to have a `fallback` event.

For example if we want to hide or disable the copy button we have an event handler to do so.

**Comment by JamesMGreene** [comment]
Related: #356

**Comment by bmintz** [comment]
"[for example if we want to hide or disable the copy button we have an event handler to do so.](https://github.com/zeroclipboard/zeroclipboard/issues/552#issuecomment-77400029)"
@tarebyte, by "we" do you mean the consumer? I would not want a _built-in_ way of hiding or disabling the "Copy" button, however, the _consumer_ doing that is totally fine, IMO.

Also, would a "copy" event still be emitted, even if the user has no Flash?
I think it should, so that if needed the consumer can still implement their own "highlight with instructions" functionality, _a la_ <http://goo.gl>.

**Comment by tarebyte** [comment]
> by "we" do you mean the consumer?

Yes the consumer, sorry about the confusion, hiding elements is not the job of zeroclipboard.

**Comment by dandv** [comment]
While I agree that ZeroClipboard shouldn't alter the UI, I've wasted ([again](https://github.com/zeroclipboard/zeroclipboard/pull/586)) a good hour trying to figure out why ZC wasn't working in IE. Turns out it wasn't IE's fault; I simply didn't have Flash installed in it!

However, there was **no error message in the console whatsoever** to help debug this. The code I used is the example from the README. I think one line in the README about the pitfall of Flash not being install would be a great compromise between some fallbacks suggested here and doing absolutely nothing.

**Comment by dandv** [comment]
Also, how does GitHub do it for gists? https://gist.github.com/jonrohan/81085b119d16cdd7868a shows the copy buttons in IE and FF without Flash, and both work.

**Comment by GeoffTidey** [comment]
I'd imagine a lot of people are ditching flash after these recent security revelations - http://www.theverge.com/2015/7/8/8911077/adobe-flash-hacking-team-vulnerability
